### PR TITLE
mat3, mat4 stuff

### PR DESCRIPTION
**Pull request by guzba**

### CODE REVIEW

File: src/vmath.nim [hunk 1]
```diff
@@ -507,44 +507,32 @@ type Mat3* = array[9, float32] ## 3x3 Matrix
...
-  result[2] = a[6]
-  result[3] = a[1]
-  result[4] = a[4]
-  result[5] = a[7]
-  result[6] = a[2]
-  result[7] = a[5]
-  result[8] = a[8]
+proc transpose*(a: Mat3): Mat3 {.inline.} =
+  [
+    a[0], a[3], a[6],
```
Comment (treeform) [inline src/vmath.nim:532]: I think using the new [0, 0] would make this even more clear.

File: src/vmath.nim [hunk 2]
```diff
@@ -850,66 +796,66 @@ proc `*`*(a, b: Mat4): Mat4 =
...
   a[14] = b.z
 
-proc rotationOnly*(a: Mat4): Mat4 =
+proc rotationOnly*(a: Mat4): Mat4 {.inline.} =
   result = a
   result.pos = vec3(0, 0, 0)
 
 proc dist*(a, b: Mat4): float32 =
-  var
+  let
```
Comment (treeform) [inline src/vmath.nim:854]: I think here the `dist(a.pos, b.pos)` might be pretty short.

File: src/vmath.nim [hunk 3]
```diff
@@ -562,77 +550,67 @@ proc `*`*(a: Mat3, b: Mat3): Mat3 =
...
+
+proc rotationMat3*(angle: float32): Mat3 {.inline.} =
   # Create a matrix from an angle.
   let
     sin = sin(angle)
     cos = cos(angle)
-  result[0, 0] = cos
-  result[0, 1] = -sin
-  result[0, 2] = 0
+  result = [
```
Comment (treeform) [inline src/vmath.nim:586]: I really like this, came out pretty well.

File: src/vmath.nim (hunk 1 above)
Comment (treeform) [reply src/vmath.nim:532]: But it can be done later...

**Review by treeform — APPROVED** [review]

File: src/vmath.nim (hunk 1 above)
Comment (guzba) [reply src/vmath.nim]: benchmarked using them, makes no perf diff so switched

**Pull request merged**